### PR TITLE
(feat): Add JS suport for instagram gallery

### PR DESCRIPTION
**Pull request by iranzo**
## Prerequisites

- [x] My Pull Request is filled against the `next` branch
- [x] All commits in my Pull Request conform to [Elegant Git commit Guidelines](https://elegant.oncrashreboot.com/git-commit-guidelines)

## Recommended Steps

- [x] My patch adds a new feature, therefore I have also added a help article about it
- [x] My patch changes Elegant behavior, therefore I have updated the help article to reflect this change
- [x] My commits are [signed](https://help.github.com/en/articles/signing-commits)

## Description

This PR modifies article.html so that it can show instagram pictures or group of pictures as galleries via adding a new yaml preamble named 'instagram'

Example at: https://deploy-preview-560--pelicanelegant.netlify.com/how-to-use-galleriaio-plugin

### CODE REVIEW

**Comment by iranzo** [comment]
Related with #385

File: documentation/content/Supported Plugins/galleriaio.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+[TOC]
+
+Similar to the use case defined in [Photogallery]({filename}photogallery.md), Pelican uses <https://galleria.io> code to show images from instagram.
+
+## Article Metadata
+
+To enable the plugin, just set a `yaml preamble` entry for instagram like:
+
+```yaml
+instagram: <picture ID>
```
Comment (talha131) [inline documentation/content/Supported Plugins/galleriaio.md]: Is it a picture ID or a gallery ID?

File: documentation/content/Supported Plugins/galleriaio.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Title: Creating a Photo Gallery Article from Instagram post
+Tags: pelican-theme, pelican-plugin, photo gallery, instagram, galleriaio
+Category: Supported Plugins
+Date: 2020-01-27 17:40
+Slug: how-to-use-galleriaio-plugin
+Subtitle:
+Summary: Elegant can be configured to provide a simple display of pictures.
+Keywords: photos, gallery, photogallery, instagram, galleriaio
+Authors: Pablo Iranzo Gómez
+instagram: B3ePmL7gd1P
```
[the remark this answers is not in the document]
Comment (iranzo) [reply documentation/content/Supported Plugins/galleriaio.md]: ok

File: templates/article.html [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {% extends 'base.html' %}
 
 {% block title %}
-{{ article.title|striptags|e }} {%if article.subtitle %} - {{ article.subtitle|striptags|e }} {% endif %} · {{ super() }}
+{{ article.title|striptags|e }} {% if article.subtitle %} - {{ article.subtitle|striptags|e }} {% endif %} · {{ super() }}
```
[the remark this answers is not in the document]
Comment (iranzo) [reply templates/article.html]: ok!

File: templates/article.html [hunk 4]
```diff
@@ -12,7 +12,7 @@
 
 {% block meta_tags_in_head %}
 {% if article.redirect %}
-<meta http-equiv="refresh" content="0;URL={{ article.redirect}}" />
+<meta http-equiv="refresh" content="0;URL={{ article.redirect }}" />
```
[the remark this answers is not in the document]
Comment (iranzo) [reply templates/article.html]: ok

File: templates/article.html [hunk 5]
```diff
@@ -23,6 +23,22 @@
 {{ smo_metadata(article) }}
 {% endblock meta_tags_in_head %}
 
+{% if article.instagram %}
```
[the remark this answers is not in the document]
Comment (iranzo) [reply templates/article.html]: I'll prefer you to handle this part in a later commit if you don't mind

File: documentation/content/Supported Plugins/galleriaio.md (hunk 1 above)
Comment (iranzo) [reply documentation/content/Supported Plugins/galleriaio.md]: As described in the doc, it's the part in the url, either a single picture or a set of (max 10 pictures I think) that gets shown, the javascript handles both and prefills the list of pictures

**Comment by talha131** [comment]
@iranzo Kindly fix the issues I have mentioned.

Next, this patch adds one more dependency to the jquery. We can use other JS frameworks that do not depend on jquery, like

1. https://photoswipe.com/
1. https://sachinchoolur.github.io/

Will you be able to replace galleriajs with photoswipe? If not then we both can work together on this feature. It ain't difficult. We can finish it in two days.

File: templates/article.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (talha131) [reply templates/article.html]: Will do.

File: documentation/content/Supported Plugins/galleriaio.md (hunk 1 above)
Comment (talha131) [reply documentation/content/Supported Plugins/galleriaio.md]: Can you please add this information to the doc, that link can be either to a single image or a bunch of images.

**Comment by iranzo** [comment]
> @iranzo Kindly fix the issues I have mentioned.
> 
> Next, this patch adds one more dependency to the jquery. We can use other JS frameworks that do not depend on jquery, like
> 
> 1. https://photoswipe.com/
> 2. https://sachinchoolur.github.io/
> 
> Will you be able to replace galleriajs with photoswipe? If not then we both can work together on this feature. It ain't difficult. We can finish it in two days.

All other things addressed.
2nd url doesn't load, for 1st one it looks nice, but to get this kind of automation (just instragram_id) I will need to check if it's possible and how.

If we get both supported, we can make something that is compatible with both so that end user can decide or us put default of our preferred one and deprecate the other (if we want to get this included asap)

**Comment by talha131** [comment]
> 2nd url doesn't load

Sorry. Here you go https://sachinchoolur.github.io/lightgallery.js/

> If we get both supported, we can make something that is compatible with both so that end user can decide or us put default of our preferred one and deprecate the other (if we want to get this included asap)

Too much work. I say as we have already decided to drop jquery, we should only choose a solution that works without jquery.

>  but to get this kind of automation (just instragram_id) I will need to check if it's possible and how.

It won't be much difficult to refactor jquery code to plain JS and make it work with photoswipe. As I said, we can work together on this feature.

----

Meanwhile, do fix the issues in this PR that I have pointed out. Photoswipe change will be made on top of this PR.

**Comment by iranzo** [comment]
One concern on ligthtgallery:

image

Is listed as free and open source library and repo contains: https://github.com/sachinchoolur/lightgallery.js/blob/master/LICENSE.md so it should be ok for us to use it

**Comment by talha131** [comment]
Right. I noticed the license.

Can you please compare both the libraries and let me know which one you think is better suited for us?
Like development activity, community etc.

**Comment by iranzo** [comment]
Based on latest posts and amount of contributors, photoswipe seems the most maintained (lightgallery is 1 year back in time and 10 contribs vs 42 in photoswipe)

**Comment by talha131** [comment]
Ok so photoswipe it is. 

Now, if you can do this task alone then please update the PR.
Else if you need my help then,

1. Resolve the issues I pointed out in the review
1. I will open a new branch so that we both can work together on it.

**Comment by iranzo** [comment]
@talha131 the ones about galleria.io are already in this PR and in the preview (together with admonition usage for the single-multiple)

**Comment by talha131** [comment]
Thank you @iranzo.

You can view my progress and changes here. https://github.com/Pelican-Elegant/elegant/pull/567

To make or add your changes open your PR against `feat-image-gallery` branch.

---
Closing this PR in favor of https://github.com/Pelican-Elegant/elegant/pull/567

**Pull request closed without merging**